llama-index-integrations/llms/llama-index-llms-cleanlab/llama_index/llms/cleanlab/base.py

```diff
@@ -0,0 +1,73 @@
+from typing import Any, Dict, Optional
+import json
+
+# Import LlamaIndex dependencies
+from llama_index.core.base.llms.types import (
+    CompletionResponse,
+    CompletionResponseGen,
+    LLMMetadata,
+)
+from llama_index.core.base.llms.generic_utils import (
+    get_from_param_or_env,
+)
+from llama_index.core.llms.callbacks import llm_completion_callback
+from llama_index.core.llms.custom import CustomLLM
+from llama_index.core.bridge.pydantic import Field
+
+from cleanlab_studio import Studio
+
+class CleanlabTLM(CustomLLM):
+    
+    # TODO: figure context_window from the underlying model (GPT-3.5 has 16k, GPT-4 has 128k)
+    context_window: int = 16000
+    max_tokens: int = 512
+    model: str = "TLM"
+
+    def __init__(
+        self,
+        api_key: Optional[str] = None,
+        quality_preset: Optional[str] = "medium",
+        additional_kwargs: Optional[Dict[str, Any]] = None,
+    ) -> None:
+        """Initialize params."""
+        additional_kwargs = additional_kwargs or {}
+
+        api_key = get_from_param_or_env("api_key", api_key, "CLEANLAB_API_KEY")
+        #self.quality_preset = quality_preset
+
```

Comment:
You should include `CallbackManager`. You can find a reference [here](https://github.com/run-llama/llama_index/blob/554bfaef06e9784354e7dc46cf1c6d9c97985845/llama-index-integrations/llms/llama-index-llms-llama-api/llama_index/llms/llama_api/base.py#L76).

Reply:
thanks!